### PR TITLE
added failing XmlEncoder testcase regarding ampersand

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/phpbench/phpbench/issues/818

requires https://github.com/phpbench/dom/pull/3

### CODE REVIEW

**Comment by staabm** [comment]
hmm it seems this now reproduces my problem, but I guess the fix should actually be in https://github.com/phpbench/dom ?

**Comment by dantleech** [comment]
> hmm it seems this now reproduces my problem, but I guess the fix should actually be

yes - looks like there needs to be another API method `createTextElement` ([see here](https://www.php.net/manual/en/domdocument.createelement.php#refsect1-domdocument.createelement-notes)).

Alternatively could refactor to use `$errorsEl->appendChild($dom->createTextElement('foo', 'bar'))` instead, but would be more consistent to add a new method to `phpbench/dom`

**Comment by dantleech** [comment]
Tagged and bumped the dom lib, so rebase should make this green.

File: lib/Serializer/XmlEncoder.php [hunk 1]
```diff
@@ -61,10 +61,9 @@ public function encode(SuiteCollection $suiteCollection): Document
                 $infoEl = $envEl->appendElement($information->getName());
 
                 foreach ($information as $key => $value) {
-                    $valueEl = $infoEl->appendElement('value');
+                    $valueEl = $infoEl->appendTextNode('value', $value);
```
Comment (dantleech) [inline lib/Serializer/XmlEncoder.php:64]: note this is just one place - in your original example I think the `echo "Foo & Bar"` was in the error message. As none of what we do here should be passing a literal `&` is is probably fine to replace all `appendElement` with `appendTextNode`

Comment (staabm) [reply lib/Serializer/XmlEncoder.php:64]: hmm I did not spot any other occurence where a node is appended with a value.

all other remaining spots seem to only set attributes. did I miss something?

Comment (staabm) [reply lib/Serializer/XmlEncoder.php:64]: ahh just saw 2 other spots. anything left?

**Comment by dantleech** [comment]
Great, thanks for this!

**Pull request merged**